src/scripts/main.js

```diff
@@ -1,3 +1,270 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const head = document.querySelectorAll('thead th');
+
+const callState = {
+  countName: 0,
+  countPosition: 0,
+  countAge: 0,
+  countOffice: 0,
+  countSalary: 0,
+};
+
+function toValidNum(param) {
+  return param.replace('$', '').replace(/,/g, '');
+}
+
+tbody.addEventListener('click', (events) => {
+  const targetRow = events.target.closest('tr');
+
+  if (!targetRow) {
+    return;
+  }
+
+  tbody.querySelectorAll('tr').forEach((el) => el.classList.remove('active'));
+  targetRow.classList.add('active');
+});
+
+head.forEach((elements) => {
+  elements.onclick = function (events) {
+    const target = events.target;
+
+    switch (target.textContent) {
+      case 'Name':
+        nameSort();
+        inerCaseCounter('countName');
```

Comment:
The function name `inerCaseCounter` seems to be a typo. It should probably be `innerCaseCounter` to reflect its purpose more accurately.